### PR TITLE
doc/dev/crimson: making it more user-friendly, re-arranging chapters and fixing typos

### PR DESCRIPTION
**Pull request by gitkenan**
This began as just fixing a few small things, however it was then recommended to me that I make any other changes I felt could be appropriate, since I am probably the target audience for this kind of document - a new user/developer of Crimson. A summary of the changes are as follows:

- Moving the section about setting essential flags which activate Crimson to its own place
- Making the commands safer and more appropriate; in particular the cmake and vstart commands
- Re-arranging the chapters to flow in order of the user's actions in setting up; that is, building Ceph for Crimson, starting the cluster, then setting the flags, then learning about extras and Crimson in general
- Fixing some typos   
- Emphasising the danger of activating Crimson in its own 'note' box for extra highlighting

I've added some comments to the 'files changed' section in an attempt to speed up the process of review. Blocks of text that have been rearranged _and_ edited can be very messy to review, so hopefully the in-line comments will help with that. 

Thanks!

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by zdover23** [comment]
@zdover23, review this in twelve hours.

File: doc/dev/crimson/crimson.rst
Comment (gitkenan) [inline doc/dev/crimson/crimson.rst:67]: This was under the 'testing crimson with cephadm' chapter. I've moved them out of there completely and put it after the vstart chapter since in practice, these commands don't work until after you've got a cluster running, so I've re-arranged the document to first include config/setup content, then the document moves on to include these instructions.

File: doc/dev/crimson/crimson.rst
Comment (gitkenan) [inline doc/dev/crimson/crimson.rst:158]: A little bit of sneaky editing here; I've re-arranged the arguments so that the object store types are all together, and also I've put vstart before the section about the object store backend for reasons mentioned in the comment above, so I've included a link to the chapter here.

File: doc/dev/crimson/crimson.rst
Comment (gitkenan) [inline doc/dev/crimson/crimson.rst:207]: The end of this line was edited to remove the cephadm exclusivity.

File: doc/dev/crimson/crimson.rst
Comment (gitkenan) [inline doc/dev/crimson/crimson.rst:219]: I've added this which will appear as a blue box on the documentation, it seemed to me like these boxes have the aesthetic affect of being like an important warning, so it seemed to be more than appropriate for this section since it's such an emphasised warning.

File: doc/dev/crimson/crimson.rst
Comment (gitkenan) [inline doc/dev/crimson/crimson.rst:509]: This apostrophe shows up in the documentation output text, meaning it's a typo.

File: doc/dev/crimson/crimson.rst
Comment (gitkenan) [inline doc/dev/crimson/crimson.rst:19]: This line is removed completely since the do_make.sh script is in /ceph/ and there's therefore no need to cd into build in order to do the next line.

File: doc/dev/crimson/crimson.rst
Comment (anthonyeleven) [inline doc/dev/crimson/crimson.rst:139]: s/the//

File: doc/dev/crimson/crimson.rst
Comment (anthonyeleven) [inline doc/dev/crimson/crimson.rst:157]: back end

File: doc/dev/crimson/crimson.rst
Comment (gitkenan) [reply doc/dev/crimson/crimson.rst:139]: I didn't change this part, could you be a bit more specific please so I can be sure to update the relevant part correctly - I see a double // i. your comment

File: doc/dev/crimson/crimson.rst
Comment (gitkenan) [reply doc/dev/crimson/crimson.rst:157]: Thanks for the review btw 🙂

File: doc/dev/crimson/crimson.rst
Comment (Matan-B) [inline doc/dev/crimson/crimson.rst:201]: Let's move this chapter to appear after `CONFIGURE CRIMSON WITH BLUESTORE`.

The order should be:
* BUILDING CRIMSON
* TESTING CRIMSON WITH CEPHADM
* CONFIGURE CRIMSON WITH BLUESTORE
* RUNNING CRIMSON
    * ENABLING CRIMSON
    * VSTART.SH
    * OBJECT STORE BACKEND

While the rest can stay the same..

What do you think?

File: doc/dev/crimson/crimson.rst
Comment (Matan-B) [inline doc/dev/crimson/crimson.rst:203]: We can add a note here that the `vstart.sh` script automatically sets these flags so this step could be skipped if vstart is used (same as the note in CONFIGURE CRIMSON WITH BLUESTORE)

Note: vstart only sets `set-allow-crimson` for now which allows the OSDs to come up.
The other 2 commands should be also added to the script in the future.

**Review by Matan-B — COMMENTED** [review]
Thank you for the suggestions!
Left a few comments.

File: doc/dev/crimson/crimson.rst
Comment (zdover23) [reply doc/dev/crimson/crimson.rst:157]: @gitkenan,
It depends on whether it's a noun (and if it is a noun, then it's "back end") or it's an adjective (and if it's an adjective, then it's "backend"). I know that this is extremely finicky, but that's English. Don't worry too much about it. We'll correct any nounal or adjectival errors later. I appreciate your work on this. It's exactly the kind of thing that I want to encourage.

I should introduce myself. I'm Zac, the upstream Ceph docs guy. You're welcome to write to me on Slack any time for help with docs-related things. I should be awake today through most of the Manchester-time-zone workday.

File: doc/dev/crimson/crimson.rst
Comment (Matan-B) [inline doc/dev/crimson/crimson.rst:186]: :+1:

File: doc/dev/crimson/crimson.rst
Comment (gitkenan) [reply doc/dev/crimson/crimson.rst:157]: Thanks very much! I appreciate that a lot, since it'll require careful sifting through the document, I'll leave that change out of this one as you've suggested. 

Great to meet you! I'm Kenan - I'm a new Ceph developer working on introducing some new plugins, but I'm particularly interested in Crimson at the moment in my spare time (I am working at IBM but I'm doing extra stuff outside of work)

Comment (gitkenan) [reply doc/dev/crimson/crimson.rst:157]: btw I'm also based in Manchester UK! 😄

File: doc/dev/crimson/crimson.rst
Comment (gitkenan) [reply doc/dev/crimson/crimson.rst:201]: Sounds great! :)

File: doc/dev/crimson/crimson.rst
Comment (gitkenan) [reply doc/dev/crimson/crimson.rst:218]: Awesome - it fits a lot better there thanks!

File: doc/dev/crimson/crimson.rst
Comment (gitkenan) [reply doc/dev/crimson/crimson.rst:203]: Okay thanks - I'm not entirely sure then why the whole step can be skipped since vstart doesn't currently do the first or third commands - should we not mention that detail so people can manually do the first and third commands in the case of running vstart until the vstart script is updated to include them?

File: doc/dev/crimson/crimson.rst
Comment (gitkenan) [inline doc/dev/crimson/crimson.rst:147]: The below diff is very ugly to follow, I recommend checking out the rich diff instead

**Review by gitkenan — COMMENTED** [review]
I think the main point of contention left here is the comment to clarify the correct usage of vstart, starting at 'running crimson'. Judging by @Matan-B's comments I've decided to just write "(this step can be replaced by adding a flag if and when you run `vstart.sh`_):" before the three commands, and left the vstart section the same assuming that the primary example of how to use vstart is correct, and there's no need to repeat commands 1 and 3 from the essential flags, there's no need for further explaining I guess.

File: doc/dev/crimson/crimson.rst
Comment (Matan-B) [reply doc/dev/crimson/crimson.rst]: Comment edited, not relevant anymore.

File: doc/dev/crimson/crimson.rst
Comment (anthonyeleven) [reply doc/dev/crimson/crimson.rst:183]: We have a convention in docs of writing "back end" as two words.  I personally prefer one, but such is the party line.  Here and below.

File: doc/dev/crimson/crimson.rst
Comment (anthonyeleven) [inline doc/dev/crimson/crimson.rst:227]: Capitalize section headers please

File: doc/dev/crimson/crimson.rst
Comment (Matan-B) [inline doc/dev/crimson/crimson.rst:129]: nit: Indentation is not accurate here. You can find the rendered docs here: https://ceph--58705.org.readthedocs.build/en/58705/dev/crimson/crimson/#running-crimson

```
.. note::
    Crimson is in a tech preview stage. 
    As you might expect, Crimson does not yet have as extensive a feature set as does ceph-osd. 
    Malfunctions including crashes and data loss are to be expected. 
```

File: doc/dev/crimson/crimson.rst
Comment (Matan-B) [inline doc/dev/crimson/crimson.rst:136]: Let's add this as a note:
```
.. note::
    `vstart.sh`_ enables crimson automatically when `--crimson` is used. 
```

File: doc/dev/crimson/crimson.rst
Comment (anthonyeleven) [reply doc/dev/crimson/crimson.rst:183]: I defer to Zac

File: doc/dev/crimson/crimson.rst
Comment (gitkenan) [reply doc/dev/crimson/crimson.rst]: Thanks 😄 Done

**Comment by zdover23** [comment]
> I think the main point of contention left here is the comment to clarify the correct usage of vstart, starting at 'running crimson'. Judging by @Matan-B's comments I've decided to just write "(this step can be replaced by adding a flag if and when you run `vstart.sh`_):" before the three commands, and left the vstart section the same assuming that the primary example of how to use vstart is correct, and there's no need to repeat commands 1 and 3 from the essential flags, there's no need for further explaining I guess.

@gitkenan, When Matan has given this PR an approval, go ahead an merge it. If we need to make more alterations in the future, we can just raise new pull requests. This is already a pretty beefy PR.

**Comment by gitkenan** [comment]
> > I think the main point of contention left here is the comment to clarify the correct usage of vstart, starting at 'running crimson'. Judging by @Matan-B's comments I've decided to just write "(this step can be replaced by adding a flag if and when you run `vstart.sh`_):" before the three commands, and left the vstart section the same assuming that the primary example of how to use vstart is correct, and there's no need to repeat commands 1 and 3 from the essential flags, there's no need for further explaining I guess.
> 
> @gitkenan, When Matan has given this PR an approval, go ahead and merge it. If we need to make more alterations in the future, we can just raise new pull requests. This is already a pretty beefy PR.

Thanks so much for the input, Matan has told me over slack that actually the above is not an issue, because vstart _does_ implement all 3 of these essential flag commands.

File: doc/dev/crimson/crimson.rst
Comment (anthonyeleven) [inline doc/dev/crimson/crimson.rst:269]: s/the application/processes/

Comment (gitkenan) [reply doc/dev/crimson/crimson.rst:269]: Thanks! Is it okay if I group this change with the s/backend/back end/ changes to avoid this PR getting too big or would you like it to be put into this one?

**Review by Matan-B — APPROVED** [review]

**Comment by zdover23** [comment]
@Matan-B, this gets backported only to Squid, right?

**Comment by Matan-B** [comment]
> @Matan-B, this gets backported only to Squid, right?

Right

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/58743 - Squid backport
No Reef backport
No Quincy backport

**Pull request merged**
